content/en/docs/concepts/services-networking/dual-stack.md

```diff
@@ -68,47 +72,162 @@ An example of an IPv6 CIDR: `fdXY:IJKL:MNOP:15::/64` (this shows the format but
+{{< codenew file="service/networking/dual-stack-default-svc.yaml" >}}
 
+   You can validate this behavior by using kubectl to inspect an existing headless service with selectors.
+
+```yaml
+kubectl get svc my-service -o yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: MyApp
+  name: my-service
+spec:
+  clusterIP: None
+  clusterIPs:
+  - None
+  ipFamilies:
+  - IPv4
+  ipFamilyPolicy: SingleStack
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: MyApp
+```
+
+#### Switching Services between single-stack and dual-stack
+
+Services can be changed from single-stack to dual-stack and from dual-stack to single-stack.
+
+   1. To change a Service from single-stack to dual-stack, change `.spec.ipFamilyPolicy` from `SingleStack` to `PreferDualStack` or `RequireDualStack` as desired. When you change this Service from single-stack to dual-stack, Kubernetes assigns the missing address family so that the Service now has IPv4 and IPv6 addresses.
+
+   Edit the Service specification updating the `.spec.ipFamilyPolicy` from `SingleStack` to `PreferDualStack`.
+
+Before:
+```yaml
+spec:
+  ipFamilyPolicy: SingleStack
+```
+After:
+```yaml
+spec:
+  ipFamilyPolicy: PreferDualStack
+```
+
+   2. To change a Service from dual-stack to single-stack, change `.spec.ipFamilyPolicy` from `PreferDualStack` or `RequireDualStack` to `SingleStack`. When you change this Service from dual-stack to single-stack, Kubernetes retains only the first element in the `.spec.ClusterIPs` array, and sets `.spec.ClusterIP` to that IP address and sets `.spec.ipFamilies` to the address family of `.spec.ClusterIPs`.
+
+### Headless Services without selector
+
+For [Headless Services without selectors](/docs/concepts/services-networking/service/#without-selectors) and without `.spec.ipFamilyPolicy` explicitly set, the `.spec.ipFamilyPolicy` field defaults to `RequireDualStack`.
+
+### Service Type LoadBalancer
+
+To provision a dual-stack LoadBalancer for your Service:
+   * Set the `.spec.type` field to `LoadBalancer`
+   * Set `.spec.ipFamilyPolicy` field to `PreferDualStack` or `RequireDualStack`
 
+{{< note >}}
+To use a dual-stack `LoadBalancer` type Service, your cloud provider must support IPv4 and IPv6 load balancers.
+{{< /note >}}
 
-## {{% heading "whatsnext" %}}
+## Egress Traffic
 
+Either publicly routable or non-publicly routable IPv6 address blocks will work if the underlying {{< glossary_tooltip text="CNI" term_id="cni" >}} provider implements the transport. If you have a Pod that uses non-publicly routable IPv6 addresses and want that Pod to reach off-cluster destinations (eg. the public Internet), you must set up IP masquerading for the egress traffic, or a similar mechanism such as transparent proxying. The [ip-masq-agent](https://github.com/kubernetes-sigs/ip-masq-agent) project supports dual-stack for IP masquerading on dual-stack clusters.
```

Comment:
nit: What are the other options that "work" besides publicly or non-publicly routable?
In other words, could you say, "all routable IPv6 address blocks are routed if the underlying ..."

Reply:
Okay, have clarified and condensed this point. Thanks.